pkgs/applications/audio/littlegptracker/default.nix

```diff
@@ -0,0 +1,81 @@
+{ stdenv
+, fetchFromGitHub
+, SDL
+, jack2
+, Foundation
+}:
+
+stdenv.mkDerivation rec {
+  pname = "littlegptracker";
+  version = "unstable-2019-04-14";
+
+  src = fetchFromGitHub {
+    owner = "Mdashdotdashn";
+    repo = "littlegptracker";
+    rev = "0ed729b46739e3df5e111c6fa4d548fde2d3b891";
+    sha256 = "1pc6lg2qp6xh7ahs5d5pb63ms4h2dz7ryp3c7mci4g37gbwbsj5b";
+  };
+
+  buildInputs = [
+    SDL
+    jack2
+  ] ++ stdenv.lib.optional stdenv.isDarwin Foundation;
+
+  postPatch = ''
+    # Remove coherency checks >:)
+    # As far as I can see, they are just to make sure that the types can fit
+    # in the word length, but this check was written when 64 bits weren't a
+    # possibility and didn't take that into account,
+    # so although the types do fit, the checks fail.
+    # Indeed, the program seems to work well on 64 bits.
+    # More info here:
+    # https://github.com/Mdashdotdashn/LittleGPTracker/issues/4
+    sed -i '126,147d' sources/Externals/Soundfont/DATATYPE.H
+
+    # Set the initial directory to the current directory, otherwise the user has
+    # to navigate from the nix store, which makes the program crash due to its
+    # size
+    sed -i 's@"bin:\.\."@"."@' sources/Adapters/DEB/System/DEBSystem.cpp
+  '';
+
+  preBuild = "cd projects";
+
+  makeFlags =
+    stdenv.lib.optionals stdenv.isLinux  [ "PLATFORM=DEB" ] ++
+    stdenv.lib.optionals stdenv.isDarwin [
+      "PLATFORM=OSX"
+      "CXX=${stdenv.cc.targetPrefix}cc" # use clang, not g++
+    ] ;
+
+  NIX_CFLAGS_COMPILE = [ "-fpermissive" ] ++
+    stdenv.lib.optional stdenv.hostPlatform.isAarch64 "-Wno-error=narrowing";
+
+  installPhase = let extension = if stdenv.isDarwin then "app" else "deb-exe";
+    in "install -Dm555 lgpt.${extension} $out/bin/lgpt";
+
+  meta = with stdenv.lib; {
+    description = "A music tracker similar to lsdj optimised to run on portable game consoles";
+    longDescription = ''
+      LittleGPTracker (a.k.a 'The piggy', 'lgpt') is a music tracker optimised
+      to run on portable game consoles. It is currently running on Game Park's
+      GP2x & Caanoo, PSP, Dingoo, Windows, Mac OSX (intel/ppc) & Linux (Debian).
+
+      It implements the user interface of littlesounddj, a very famous tracker
+      for the Gameboy platform that has been tried and tested by many users over
+      the years, leading to a little complex but yet extremely efficent way of
+      working.
+
+      Piggy currently supports 8 monophonic 16Bit/44.1Khz stereo sample playback
+      channels. Additionally, the program can drive MIDI instruments (with the
+      gp32 and gp2x a custom MIDI interface is required).
+    '';
+    homepage = "http://www.littlegptracker.com/";
+    downloadPage = "http://www.littlegptracker.com/download.php";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ fgaz ];
+    platforms = platforms.all;
+    # Cannot recognize Foundation, help appreciated
+    broken = stdenv.isDarwin;
```

Comment:
`projects/osx_rules`'s linker invocation doesn't look for the Foundation framework:

```make
$(CXX) $(LDFLAGS) -arch i386 -framework SDL -framework Cocoa -framework Carbon -o $@ $(OFILES) $(LIBS)
```

Looking through Nixpkgs, maybe this will help?

```nix
NIX_LDFLAGS = stdenv.lib.optionalString stdenv.isDarwin "-framework CoreFoundation";
```

Reply:
Really helpful, thank you! Let's see if it works..

Reply:
Sorry, `-framework Foundation` not `CoreFoundation`. 2 different Darwin packages it seems…

Although looking at the CI log, it seems to be unable to find the *headers*, so those LDFLAGS prolly won't change a thing at this point. I don't know a thing about Apple LLVM and Darwin Frameworks…

Maybe adding it to the CFLAGS might help? If not, you could try adding an include path into that framework instead. Something like `-isystem ${Foundation}/Library/Frameworks/Foundation.framework/Headers`? Just guesstimating, don't have a Darwin machine to test either.